### ISSUE TITLE
Backspace not working

### ISSUE THREAD

**Issue by aldoridhoni** [rbanffy/selectric-mode#20]
After enabling selectric-mode in spacemacs, `Backspace` button is not working.

**Comment by MD-ThorErik** [comment]
Sometimes backspace makes the correct sound but moves cursor back a character instead of deleting.
Other times, it says "List nesting exceeds 'max-lisp-eval-depth'" when pressing Backspace repeteadly, cursor does not move and the sound is... wrong.

**Comment by bkruczyk** [comment]
I seems fixed, I checked selectric-mode in newest spacemacs and `Backspace` is fine, you can hear sound and no warnings are showing in `Messages` buffer.

**Comment by MD-ThorErik** [comment]
No, it's not fixed. I just start spacemacs, switch to scratch buffer, start selectric mode with <space>-C-t-t and type (I use vim editing style). Sometimes backspace makes the sound and moves the cursor back but doen't delete a character. Some other times it makes a clipped sound, the cursor doesn't move and get the error message "nesting exceeds `max-lisp-eval-depth'".
Some other times it works when enabled with M-x 'selectric-mode' in insert mode.

**Comment by bkruczyk** [comment]
@MD-ThorErik I don't know which version of Spacemacs you use, but backspace is working just fine with `selectric-mode` on current spacemacs-master.

At least on *my* computer :)

**Comment by MD-ThorErik** [comment]
It doesn't work in master or develop. My config is not the out-of-the-box one, evidently. I have many packages too.
I think it's something in the `selectric-move-sound` function because the error happens when deleting or when moving the cursor. Still, I don't quite know how to debug it...

**Comment by bkruczyk** [comment]
@MD-ThorErik upload your config or maybe even whole `.emacs.d` if you can do that. Maybe I can figure something out.

**Comment by MD-ThorErik** [comment]
Well, I deleted emacs.d and .spacemacs and created them again. The issue persists.

Patching selectric.el with (#21) solves all of this.

This is my .spacemacs:
https://gist.github.com/MD-ThorErik/3352221bbd49fb72b54ad57ccd86f413

**Comment by IQubic** [comment]
So I just pulled this from package from melpa, and I want to say, Backspace isn't working in evil insert mode. Running `C-h k` in that scenario gives this:

```
DEL (translated from <backspace>) runs the command #[0 "\302 \210\303\304\300 \"!\207" ["DEL"
selectric-saved-bindings selectric-move-sound call-interactively gethash] 4 nil nil]
(found in selectric mode-map), which is an interactive compiled Lisp function.  

It is bound to DEL.
```

**Comment by tecosaur** [comment]
I'm also getting an issue with backspace just moving the cursor, also using evil mode.
